internal/backend/sessions/sessionworkflows/callbacks_misc.go

```diff
@@ -37,33 +34,55 @@ func (w *sessionWorkflow) start(wctx workflow.Context) func(context.Context, sdk
 
 		data := w.data
 
-		projectID := data.Session.ProjectID()
-		buildID := data.Session.BuildID()
+		data.Session = sdktypes.NewSession(data.Session.BuildID(), loc, inputs, memo).
+			WithParentSessionID(data.Session.ID()).
+			WithDeploymentID(data.Session.DeploymentID()).
+			WithProjectID(data.Session.ProjectID()).
+			SetDurable(data.Session.IsDurable())
 
 		if project.IsValid() {
-			var resp getProjectIDAndActiveBuildIDResponse
-			if err := workflow.ExecuteActivity(wctx, getProjectIDAndActiveBuildID, &getProjectIDAndActiveBuildIDParams{Project: project, OrgID: data.OrgID}).Get(wctx, &resp); err != nil {
+			params := getProjectIDAndActiveBuildIDParams{Project: project, OrgID: data.OrgID}
+
+			var (
+				resp *getProjectIDAndActiveBuildIDResponse
+				err  error
+			)
+
+			if inActivity {
+				resp, err = w.ws.getProjectIDAndActiveBuildIDActivity(ctx, params)
+			} else {
+				err = workflow.ExecuteActivity(wctx, getProjectIDAndActiveBuildIDActivityName, &params).Get(wctx, &resp)
+			}
+
+			if err != nil {
 				return sdktypes.InvalidSessionID, fmt.Errorf("could not get active build ID for project %s: %w", project, err)
 			}
-			buildID = resp.BuildID
-			projectID = resp.ProjectID
 
+			data.Session = data.Session.
+				WithProjectID(resp.ProjectID).
+				WithBuildID(resp.BuildID)
+
+			l = l.With(zap.Any("child_project_id", resp.ProjectID), zap.Any("child_build_id", resp.BuildID))
 		}
 
-		data.Session = sdktypes.NewSession(buildID, loc, inputs, memo).
-			WithParentSessionID(data.Session.ID()).
-			WithDeploymentID(data.Session.DeploymentID()).
-			WithProjectID(projectID).
-			SetDurable(data.Session.IsDurable())
+		var (
+			sid sdktypes.SessionID
+			err error
+		)
+
+		if inActivity {
+			sid, err = w.ws.startChildSessionActivity(ctx, data.Session)
+		} else {
+			sid, err = w.ws.StartChildWorkflow(wctx, data.Session)
+		}
 
-		sid, err := w.ws.StartChildWorkflow(wctx, data.Session)
 		if err != nil {
 			return sdktypes.InvalidSessionID, err
 		}
 
 		data.Session = data.Session.WithID(sid)
 
-		w.l.Info("child session started", zap.Any("child", sid), zap.Any("parent", w.data.Session.ID()))
+		l.Info("child session started", zap.Any("child_session", data.Session))
```

Comment:
make the logs readable

```suggestion
		l.Info("child session "+sid+" for session " + parentSessionID, zap.Any("child_session", data.Session))
```

Reply:
done